esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -759,107 +308,16 @@ def __init__(
-            )
-
-        if resolution is not None:
-            if not (isinstance(resolution, int) and resolution > 0):
-                raise ValueError("resolution must be a positive integer.")
-            if method != "conservative":
-                raise ValueError(
-                    "resolution can only be set for conservative regridding."
-                )
-        # Missing data tolerance.
-        # Code directly copied from iris.
-        if mdtol is None:
-            if method == "conservative":
-                mdtol = 1
-            elif method == "bilinear":
-                mdtol = 0
-        if not (0 <= mdtol <= 1):
-            msg = "Value for mdtol must be in range 0 - 1, got {}."
-            raise ValueError(msg.format(mdtol))
-        self.mdtol = mdtol
-        self.method = method
-        self.resolution = resolution
-
-        self.src_mask = _get_mask(src_grid_cube, use_src_mask)
-        self.tgt_mask = _get_mask(target_mesh_cube, use_tgt_mask)
-
-        partial_regrid_info = _regrid_rectilinear_to_unstructured__prepare(
-            src_grid_cube,
-            target_mesh_cube,
-            method=self.method,
+        super().__init__(
+            src,
+            tgt,
+            method,
+            mdtol=mdtol,
             precomputed_weights=precomputed_weights,
-            resolution=self.resolution,
-            src_mask=self.src_mask,
-            tgt_mask=self.tgt_mask,
-        )
-
-        # Store regrid info.
-        (
-            _,
-            _,
-            self.grid_x,
-            self.grid_y,
-            self.mesh,
-            self.location,
-            self.regridder,
-        ) = partial_regrid_info
-
-    def __call__(self, cube):
-        """
-        Regrid this :class:`~iris.cube.Cube` onto the target mesh of this regridder instance.
-
-        The given :class:`~iris.cube.Cube` must be defined with the same grid as the source
-        cube used to create this :class:`MeshToGridESMFRegridder` instance.
-
-        Parameters
-        ----------
-        cube : :class:`iris.cube.Cube`
-            A :class:`~iris.cube.Cube` instance to be regridded.
-
-        Returns
-        -------
-        :class:`iris.cube.Cube`
-            A :class:`~iris.cube.Cube` defined with the horizontal dimensions of the target
-            and the other dimensions from ``cube``. The
-            :attr:`~iris.cube.Cube.data` values of
-            ``cube`` will be converted to values on the new grid using
-            area-weighted regridding via :mod:`esmpy` generated weights.
-
-        """
-        grid_x = _get_coord(cube, "x")
-        grid_y = _get_coord(cube, "y")
-        # Ignore differences in var_name that might be caused by saving.
-        self_grid_x = copy.deepcopy(self.grid_x)
-        self_grid_x.var_name = grid_x.var_name
-        self_grid_y = copy.deepcopy(self.grid_y)
-        self_grid_y.var_name = grid_y.var_name
```

Comment:
I believe this is something I do so that regridder saving works smoother, I haven't added regridder saving capabilities yet, but it's probably worth maintaining this functionality still.

Reply:
see https://github.com/SciTools-incubator/iris-esmf-regrid/pull/198/commits/b3d43c5354cfe325214b1832ac32140301c04b04